spark/src/main/scala/geotrellis/spark/io/cassandra/CassandraDriver.scala

```diff
@@ -0,0 +1,54 @@
+package geotrellis.spark.io.cassandra
+
+import java.nio.ByteBuffer;
+
+import geotrellis.spark._
+import geotrellis.raster._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import com.datastax.spark.connector.rdd.CassandraRDD
+import com.datastax.spark.connector._
+
+import com.datastax.driver.core.DataType.{text, blob}
+import com.datastax.driver.core.schemabuilder.SchemaBuilder
+import com.datastax.driver.core.Session
+
+class TableNotFoundError(table: String) extends Exception(s"Target Cassandra table '$table' does not exist.")
+
+trait CassandraDriver[K] extends Serializable {
+  def applyFilter(rdd: CassandraRDD[(String, ByteBuffer)], layerId: LayerId, filterSet: FilterSet[K]): RDD[(String, ByteBuffer)]
+  def decode(rdd: RDD[(String, ByteBuffer)], metaData: RasterMetaData): RasterRDD[K]
+  def rowId(id: LayerId, key: K): String
+
+  def load(sc: SparkContext, keyspace: String)(
+    id: LayerId, metaData: RasterMetaData, table: String, filters: FilterSet[K]): RasterRDD[K] = {
+
+    val rdd: CassandraRDD[(String, ByteBuffer)] = 
+      sc.cassandraTable[(String, ByteBuffer)](keyspace, table).select("id", "tile")
+
+    val filteredRDD = applyFilter(rdd, id, filters)
```

Comment:
Ok, thanks for the notes Kevin, I didn't know that about the Cassandra RDDs. I'm going to merge this in and hope to see if the where filters shake out upon testing.